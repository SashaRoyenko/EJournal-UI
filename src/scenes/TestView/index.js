import React from 'react';
import { getUsers, getUserById } from '../../shared/query/test/users';

export default class TestView extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      userId: 0,
      users: [],
    };

    this.getUsers = this.getUsers.bind(this);
    this.onChangeUserId = this.onChangeUserId.bind(this);
    this.getUserById = this.getUserById.bind(this);
  }

  onChangeUserId(event) {
    this.setState({ userId: event.target.value });
  }

  async getUsers() {
    getUsers().then((users) => this.setState({ users }));
  }

  async getUserById() {
    const { userId } = this.state;
    getUserById(userId).then((user) => this.setState({ users: [user] }));
  }

  render() {
    const { users, userId } = this.state;
    return (
      <>
        <label>
          <input type="text" value={userId} onChange={this.onChangeUserId} />
        </label>
        <button onClick={this.getUserById} type="button">
          Get one user
        </button>
        <button onClick={this.getUsers} type="button">
          Get all users
        </button>
        {users.map((user, idx) => (
          <div key={user.id || idx}>
            <p>{user.firstName}</p>
            <p>{user.lastName}</p>
            <p>{user.age}</p>
            <p>{user.birthDate}</p>
          </div>
        ))}
      </>
    );
  }
}
