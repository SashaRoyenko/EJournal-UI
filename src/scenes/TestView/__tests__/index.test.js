import React from 'react';
import { mount } from 'enzyme';
import { url, getQuery } from '../../../shared/query/test/constants';
import TestView from '../index';

describe('<TestView />', () => {
  it('should render correctly', () => {
    const sut = mount(<TestView />);

    expect(sut).toMatchSnapshot();
  });

  it('should show user with 0 id', () => {
    const mockFetchPromise = Promise.resolve({
      json: () => [{ id: 1, firstName: 'Eric', lastName: 'Cartman', age: '10', birthDate: '10101010'}],
    });

    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

    const sut = mount(<TestView />);

    sut.find('button').at(1).simulate('click');

    expect(sut).toMatchSnapshot();
  });

  it('should show user with 10 id', () => {
    const valueMock = '10';
    const changeEventMock = {
      preventDefault() {},
      target: { value: valueMock }
    }
    const mockFetchPromise = Promise.resolve({
      json: () => ({ id: 1, firstName: 'Eric', lastName: 'Cartman', age: '10', birthDate: '10101010'}),
    });

    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

    const sut = mount(<TestView />);

    sut.find('input').simulate('change', changeEventMock);
    sut.find('button').at(0).simulate('click');

    expect(sut).toMatchSnapshot();

    expect(global.fetch).toHaveBeenCalledWith(`${url}users/user/${valueMock}`, getQuery); 
  });
});
