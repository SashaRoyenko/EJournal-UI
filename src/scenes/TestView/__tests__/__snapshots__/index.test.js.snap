// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestView /> should render correctly 1`] = `
<TestView>
  <label>
    <input
      onChange={[Function]}
      type="text"
      value={0}
    />
  </label>
  <button
    onClick={[Function]}
    type="button"
  >
    Get one user
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Get all users
  </button>
</TestView>
`;

exports[`<TestView /> should show user with 0 id 1`] = `
<TestView>
  <label>
    <input
      onChange={[Function]}
      type="text"
      value={0}
    />
  </label>
  <button
    onClick={[Function]}
    type="button"
  >
    Get one user
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Get all users
  </button>
</TestView>
`;

exports[`<TestView /> should show user with 10 id 1`] = `
<TestView>
  <label>
    <input
      onChange={[Function]}
      type="text"
      value="10"
    />
  </label>
  <button
    onClick={[Function]}
    type="button"
  >
    Get one user
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Get all users
  </button>
</TestView>
`;
